File: amic-platform/src/modules/fdd/types/nwc.ts
// ── Enums ────────────────────────────────────────────────

export type NWCStatus = "DRAFT" | "REVIEW" | "APPROVED";

export type NWCClassification = "ABOVE_LINE" | "BELOW_LINE" | "EXCLUDED";

export type PegMethod =
  | "LTM_AVERAGE"
  | "TTM"
  | "LAST_MONTH"
  | "MAX"
  | "MIN"
  | "CUSTOM";

// ── NWC Line Item ───────────────────────────────────────

export interface NWCLineItemRead {
  id: string;
  nwc_calculation_id: string;
  deal_id: string;
  account_code: string;
  account_name: string;
  line_item_category: string;
  classification: NWCClassification;
  amount: string; // Decimal as string — NEVER number
  monthly_amounts: Record<string, string>;
  display_order: number;
  created_at: string;
  updated_at: string;
}

export interface NWCLineItemUpdate {
  classification: NWCClassification;
}

// ── NWC Calculation ─────────────────────────────────────

export interface NWCCalculationRead {
  id: string;
  deal_id: string;
  snapshot_id: string;
  total_current_assets: string; // Decimal as string
  total_current_liabilities: string;
  net_working_capital: string;
  peg_method: PegMethod;
  peg_target: string;
  peg_delta: string;
  monthly_trend: Record<string, MonthlyTrendEntry>;
  category_breakdown: Record<string, unknown>;
  engine_version: string;
  status: NWCStatus;
  line_items: NWCLineItemRead[];
  created_at: string;
  updated_at: string;
}

export interface MonthlyTrendEntry {
  current_assets: string;
  current_liabilities: string;
  nwc: string;
}

export interface NWCRunRequest {
  snapshot_id: string;
  peg_method?: PegMethod;
  custom_peg_value?: string | null;
}

// ── NWC Summary ─────────────────────────────────────────

export interface NWCSummary {
  net_working_capital: string;
  total_current_assets: string;
  total_current_liabilities: string;
  peg_method: PegMethod;
  peg_target: string;
  peg_delta: string;
  above_line_items: NWCLineItemRead[];
  below_line_items: NWCLineItemRead[];
  is_above_target: boolean;
}

// ── Peg Simulation ──────────────────────────────────────

export interface PegSimulationRequest {
  peg_method: PegMethod;
  custom_value?: string | null;
}

export interface PegSimulationResult {
  method: PegMethod;
  target_nwc: string;
  delta: string;
  description: string;
}

export interface PegSimulationResponse {
  reference_nwc: string;
  scenarios: PegSimulationResult[];
}

File: fdd/backend/app/schemas/nwc.py
"""NWC (Net Working Capital) 관련 Pydantic 스키마 — EPIC-06."""

import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

from app.models.nwc import NWCClassification, NWCStatus, PegMethod

# -- NWC Line Item -----------------------------------------------


class NWCLineItemRead(BaseModel):
    id: uuid.UUID
    nwc_calculation_id: uuid.UUID
    deal_id: uuid.UUID
    account_code: str
    account_name: str
    line_item_category: str
    classification: NWCClassification
    amount: Decimal
    monthly_amounts: dict
    display_order: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class NWCLineItemUpdate(BaseModel):
    """WC 항목 분류 변경 요청 — FDD-601."""

    classification: NWCClassification


# -- NWC Calculation ---------------------------------------------


class NWCRunRequest(BaseModel):
    """NWC 계산 실행 요청."""

    snapshot_id: uuid.UUID
    peg_method: PegMethod = PegMethod.LTM_AVERAGE
    custom_peg_value: Decimal | None = None


class NWCCalculationRead(BaseModel):
    id: uuid.UUID
    deal_id: uuid.UUID
    snapshot_id: uuid.UUID
    total_current_assets: Decimal
    total_current_liabilities: Decimal
    net_working_capital: Decimal
    peg_method: PegMethod
    peg_target: Decimal
    peg_delta: Decimal
    monthly_trend: dict
    category_breakdown: dict
    engine_version: str
    status: NWCStatus
    line_items: list[NWCLineItemRead]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# -- NWC Peg Simulation -----------------------------------------


class PegSimulationRequest(BaseModel):
    """Peg 시나리오 시뮬레이션 요청 — FDD-603."""

    peg_method: PegMethod
    custom_value: Decimal | None = None


class PegSimulationResult(BaseModel):
    """단일 Peg 시나리오 결과."""

    method: PegMethod
    target_nwc: Decimal
    delta: Decimal
    description: str


class PegSimulationResponse(BaseModel):
    """Peg 6종 시뮬레이션 응답."""

    reference_nwc: Decimal
    scenarios: list[PegSimulationResult]


# -- NWC Summary -------------------------------------------------


class NWCSummary(BaseModel):
    """NWC 요약 (프론트엔드 표시용)."""

    net_working_capital: Decimal
    total_current_assets: Decimal
    total_current_liabilities: Decimal
    peg_method: PegMethod
    peg_target: Decimal
    peg_delta: Decimal
    above_line_items: list[NWCLineItemRead]
    below_line_items: list[NWCLineItemRead]
    is_above_target: bool
